kernel/src/main/java/org/kframework/compile/checks/CheckAtt.java

```diff
@@ -71,85 +103,109 @@ private void checkLabel(Sentence sentence) {
       if (label.contains("`") || whitespace.matcher(label).find()) {
         errors.add(
             KEMException.compilerError(
-                "Rule label '" + label + "' cannot contain whitespace or backticks.", sentence));
+                "Label '" + label + "' cannot contain whitespace or backticks.", sentence));
       }
     }
   }
 
-  private void checkUnrecognizedAtts(Sentence sentence) {
-    if (!sentence.att().unrecognizedKeys().isEmpty()) {
+  private void checkMacro(Rule rule) {
+    if (rule.isMacro()) {
+      kem.registerCompilerWarning(
+          ExceptionType.FUTURE_ERROR,
+          errors,
+          "The attribute ["
+              + rule.att().getMacro().get()
+              + "] has been deprecated on rules. Use this label on syntax declarations instead.",
+          rule);
+    }
+  }
```

Comment:
We discussed yesterday turning this into an error. Can we do that in this PR?

Reply:
I can make a follow up PR - want to get this merged now for my work on #3933

Reply:
Bruce handled it in #3956